stubs/pygit2/pygit2/_pygit2.pyi

```diff
@@ -0,0 +1,773 @@
+GIT_SUBMODULE_STATUS_WD_WD_MODIFIED: int
+LIBGIT2_VERSION: str
+LIBGIT2_VER_MAJOR: int
+LIBGIT2_VER_MINOR: int
+LIBGIT2_VER_REVISION: int
+
+_GIT_OBJ_BLOB: TypeAlias = Literal[3]
+_GIT_OBJ_COMMIT: TypeAlias = Literal[1]
+_GIT_OBJ_TAG: TypeAlias = Literal[4]
+_GIT_OBJ_TREE: TypeAlias = Literal[2]
+
+class Object:
+    _pointer: bytes
+    filemode: FileMode
+    hex: str
+    id: Oid
+    name: str | None
+    oid: Oid
+    raw_name: bytes | None
+    short_id: str
+    type: Literal[_GIT_OBJ_COMMIT, _GIT_OBJ_TREE, _GIT_OBJ_TAG, _GIT_OBJ_BLOB]
+    type_str: Literal["commit", "tree", "tag", "blob"]
+    @overload
+    def peel(self, target_type: Literal[_GIT_OBJ_COMMIT]) -> Commit: ...
+    @overload
+    def peel(self, target_type: Literal[_GIT_OBJ_TREE]) -> Tree: ...
+    @overload
+    def peel(self, target_type: Literal[_GIT_OBJ_TAG]) -> Tag: ...
+    @overload
+    def peel(self, target_type: Literal[_GIT_OBJ_BLOB]) -> Blob: ...
+    @overload
+    def peel(self, target_type: None) -> Commit | Tree | Blob: ...
+    def read_raw(self) -> bytes: ...
+    def __eq__(self, other) -> bool: ...
+    def __ge__(self, other) -> bool: ...
+    def __gt__(self, other) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __le__(self, other) -> bool: ...
+    def __lt__(self, other) -> bool: ...
+    def __ne__(self, other) -> bool: ...
+
+@final
+class Reference:
+    name: str
+    raw_name: bytes
+    raw_shorthand: bytes
+    raw_target: Oid | bytes
+    shorthand: str
+    target: Oid | str
+    type: ReferenceType
+    def __init__(self, *args) -> None: ...
+    def delete(self) -> None: ...
+    def log(self) -> Iterator[RefLogEntry]: ...
+    @overload
+    def peel(self, type: Literal[_GIT_OBJ_COMMIT]) -> Commit: ...
+    @overload
+    def peel(self, type: Literal[_GIT_OBJ_TREE]) -> Tree: ...
+    @overload
+    def peel(self, type: Literal[_GIT_OBJ_TAG]) -> Tag: ...
+    @overload
+    def peel(self, type: Literal[_GIT_OBJ_BLOB]) -> Blob: ...
+    @overload
+    def peel(self, type: None) -> Commit | Tree | Blob: ...
+    def rename(self, new_name: str) -> None: ...
+    def resolve(self) -> Reference: ...
+    def set_target(self, target: _OidArg, message: str = ...) -> None: ...
+    def __eq__(self, other) -> bool: ...
+    def __ge__(self, other) -> bool: ...
+    def __gt__(self, other) -> bool: ...
+    def __le__(self, other) -> bool: ...
+    def __lt__(self, other) -> bool: ...
+    def __ne__(self, other) -> bool: ...
+
+class AlreadyExistsError(ValueError): ...
+
+@final
+class Blob(Object):
+    data: bytes
+    is_binary: bool
+    size: int
+    def diff(self, blob: Blob = ..., flag: int = ..., old_as_path: str = ..., new_as_path: str = ...) -> Patch: ...
+    def diff_to_buffer(
+        self, buffer: bytes | None = None, flag: DiffOption = ..., old_as_path: str = ..., buffer_as_path: str = ...
+    ) -> Patch: ...
+
+# This is not a real subclassing. Just ensuring type-checkers sees this type as compatible with _CDataBase
```

Comment:
I don't understand this comment, how is `_CDataBase` involved?

Reply:
This is mostly copied from `stubs/JACK-Client/jack/__init__.pyi`, as the type checker apparently has some issues with the CFFI types. What's the current best practice of typing CFFI bindings though?